src/components/organisms/LocalSearch.tsx

```diff
@@ -0,0 +1,979 @@
+    updateSearchResults,
+  ])
+
+  const handleSearchInputKeyDown = useCallback(
+    (event: KeyboardEvent<HTMLTextAreaElement>) => {
+      switch (event.key) {
+        case 'Enter':
+          event.preventDefault()
+          event.stopPropagation()
+          if (event.ctrlKey && event.shiftKey) {
+            addNewlineToSearchValue()
+          } else {
+            navigateToNextItem(SearchResultNavigationDirection.NEXT)
+          }
+          break
+        // todo: [komediruzecki-14/02/2021] Change this key binding to CTRL+R or
+        //  or something better when codemirror shortcut gets updated
+        case 'h':
+        case 'H':
+          if (event.ctrlKey && onReplaceToggle) {
+            event.preventDefault()
+            event.stopPropagation()
+            onReplaceToggle(true)
+          }
+          break
+        case 'x':
+        case 'X':
+          if (event.altKey) {
+            event.preventDefault()
+            event.stopPropagation()
+            toggleRegexSearch()
+          }
+          break
+        case 'C':
+        case 'c':
+          if (event.altKey) {
+            event.preventDefault()
+            event.stopPropagation()
+            toggleCaseSensitiveSearch()
+          }
+          break
+        case 'Tab':
+          // Focus replace if open
+          if (showReplace) {
+            event.preventDefault()
+            event.stopPropagation()
+            setFocusingReplace(true)
+          }
+          break
+        case 'F3':
+          navigateToNextItem(
+            event.shiftKey
+              ? SearchResultNavigationDirection.PREVIOUS
+              : SearchResultNavigationDirection.NEXT
+          )
+          break
+        case 'Down': // IE/Edge specific value
+        case 'ArrowDown':
+          if (textAreaRows() <= 1) {
+            navigateToNextItem(SearchResultNavigationDirection.NEXT)
+          }
+          break
+        case 'Up': // IE/Edge specific value
+        case 'ArrowUp':
+          if (textAreaRows() <= 1) {
+            event.preventDefault()
+            event.stopPropagation()
+            navigateToNextItem(SearchResultNavigationDirection.PREVIOUS)
+          }
+          break
+        case 'Escape':
+          onSearchClose()
+          break
+        case 'F':
+        case 'f':
+          if (event.ctrlKey && onReplaceToggle) {
+            onReplaceToggle(false)
+          }
+          break
+      }
+    },
+    [
+      addNewlineToSearchValue,
+      navigateToNextItem,
+      onReplaceToggle,
+      onSearchClose,
+      showReplace,
+      textAreaRows,
+      toggleCaseSensitiveSearch,
+      toggleRegexSearch,
+    ]
+  )
+
+  useEffect(() => {
+    function onContentChange(
+      _: CodeMirror.Editor,
+      changes: EditorChangeLinkedList
+    ) {
+      if (changes.origin == '@ignore') {
+        return
+      }
+
+      /*
+       * Optimized version: looks at current line and removed text to get info about changes then
```

Comment:
What is optimized or not? I don't understand

Reply:
It's a try to optimize this function execution on each code editor change. 
This happens when user has opened search/replace and focuses the editor and starts writing something.
In that case, local search is listening for changes and updates the found items markers (and count) based on newly added/removed text.

After trying few things, only option that works in all cases is that this function runs on each change, and for now this works OK fast, if user experiences a bit of lagging (maybe huge file, a lot of matches) he should close the search to not trigger this.

 In any other way (normal usage) the search results are updated and marked in editor (works nicely).
 
 So for now we should maybe just remove all the "optimized" code and stick with update on each change (don't think this could be debounced, but maybe you have some ideas how to achieve consistency with that).
 
 The optimized versions could work if we would track state of the changes/removals (a bit tricky)